src/zen/websocket_server.cpp

```diff
@@ -0,0 +1,1360 @@
+        int len = -1;
+        try {
+            len = std::stoi(strLen);
+        } catch (const std::exception &e) {
+            LogPrint("ws", "%s():%d - %s\n", __func__, __LINE__, e.what());
+            return INVALID_PARAMETER;
+        }
+        if (len < 1)
+        {
+            LogPrint("ws", "%s():%d - invalid len %d\n", __func__, __LINE__, len);
+            return INVALID_PARAMETER;
+        }
+        if (len > MAX_BLOCKS_REQUEST)
+        {
+            LogPrint("ws", "%s():%d - invalid len %d (max is %d)\n", __func__, __LINE__, len, MAX_BLOCKS_REQUEST);
+            return INVALID_PARAMETER;
+        }
+
+        std::list<CBlockIndex*> listBlock;
+        CBlockIndex* pblockindexStart = NULL;
+        int lastH = 0;
+        for (const UniValue& o : hashes.getValues())
+        {
+            if (o.isObject())
+            {
+                LogPrint("ws", "%s():%d - invalid obj\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+            CBlockIndex* pblockindex = NULL;
+            uint256 hash(uint256S(o.get_str()));
+            {
+                LOCK(cs_main);
+                BlockMap::iterator mi = mapBlockIndex.find(hash);
+                if (mi != mapBlockIndex.end())
+                {
+                    pblockindex = (*mi).second;
+                    if (!chainActive.Contains(pblockindex))
+                        pblockindex = NULL;
+                }
+            }
+            if (pblockindex == NULL)
+            {
+                LogPrint("ws", "%s():%d - block index not found for hash[%s], skipping it\n", __func__, __LINE__, o.get_str());
+                continue;
+            }
+            if (pblockindex->nHeight > lastH)
+            {
+                lastH = pblockindex->nHeight;
+                pblockindexStart = pblockindex;
+            }
+        }
+        {
+            LOCK(cs_main);
+            CBlockIndex* nextBlock = pblockindexStart;
+            if (nextBlock == NULL)
+            {
+                LogPrint("ws", "%s():%d - start block index not found\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+            int n = 0;
+            while (n < len) {
+                listBlock.push_back(nextBlock);
+                nextBlock = chainActive.Next(nextBlock);
+                if (nextBlock == NULL)
+                    break;
+                n++;
+            }
+        }
+        sendHashes(listBlock.front()->nHeight, listBlock, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+    }
+
+    int sendRawMempool(const std::string& clientRequestId) {
+        LOCK(mempool.cs);
+
+        std::list<uint256> hashes;
+        BOOST_FOREACH(const PAIRTYPE(uint256, CTxMemPoolEntry)& entry, mempool.mapTx)
+        {
+           hashes.push_back(entry.first);
+        }
+        sendMempool(hashes.size(), hashes, WsEvent::MSG_RESPONSE, clientRequestId);
```

Comment:
Yes because the explorer compares the transactions received the websocket with the transactions ids sent and it deletes the missing transactions in the response.